jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/insights/QueryEditorDialog.kt

```diff
@@ -0,0 +1,151 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cloudwatch.logs.insights
+
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.ui.DialogWrapper
+import com.intellij.openapi.ui.ValidationInfo
+import java.awt.event.ActionEvent
+import java.util.Date
+import javax.swing.Action
+import javax.swing.JComponent
+import software.aws.toolkits.resources.message
+
+class QueryEditorDialog(
+    private val project: Project,
+    private val lGroupName: String,
+    private val absoluteTimeSelected: Boolean = false
+
+) : DialogWrapper(project) {
+    constructor(project: Project, logGroupName: String) :
+        this(project = project, lGroupName = logGroupName)
+
+    private val view = QueryEditor(project)
+    private val qQueryingLogGroupApiCall = QueryingLogGroups(project)
+    private val action: OkAction = QueryLogGroupOkAction()
+    private val validator = QueryEditorValidator()
+    init {
+        super.init()
+        title = "Query Log Groups"
+        view.absoluteTimeRadioButton.addActionListener {
+            view.qStartDate.isEnabled = true
+            view.qEndDate.isEnabled = true
+            view.RelativeTimeNumber.isEnabled = false
+            view.RelativeTimeUnit.setEnabled(false)
+        }
+        view.relativeTimeRadioButton.addActionListener {
+            view.qStartDate.isEnabled = false
+            view.qEndDate.isEnabled = false
+            view.RelativeTimeNumber.isEnabled = true
+            view.RelativeTimeUnit.setEnabled(true)
+        }
+        view.queryLogGroupsRadioButton.addActionListener {
+            view.queryBox.isEnabled = true
+            view.querySearchTerm.isEnabled = false
+        }
+        view.searchTerm.addActionListener {
+            view.queryBox.isEnabled = false
+            view.querySearchTerm.isEnabled = true
+        }
+    }
+    // qpanel is the underlying panel for the Query Editor
+    override fun createCenterPanel(): JComponent? = view.qpanel
+    override fun doValidate(): ValidationInfo? = validator.validateEditorEntries(view)
+    override fun getOKAction(): Action = action
+    override fun doCancelAction() {
+        super.doCancelAction()
+    }
+
+    override fun doOKAction() {
+        // Do nothing, close logic is handled separately
+    }
+
+    private fun getRelativeTime(unitOfTime: String, relTimeNumber: Long): StartEndDate {
```

Comment:
feels like we're reinventing the wheel here? take a look at `java.time.Instant#minus`